### PR TITLE
refactoring and fixing bunch of bugs as well as adding seek o command

### PR DESCRIPTION
**Pull request by Vane11ope**
now 'o'  command works as seek

### CODE REVIEW

**Comment by Vane11ope** [comment]
@XVilka ok fixed

File: libr/core/panels.c [hunk 1]
```diff
@@ -1644,15 +1656,26 @@ R_API int r_core_visual_panels(RCore *core, RPanels *panels) {
...
+			" 0-9  - jump to the nth panel. n is the number you hit\n"
 			);
 		r_cons_flush ();
 		r_cons_any_key (NULL);
 		break;
 	case 'b':
 		r_core_visual_browse (core);
 		break;
+	case 'o':
+		if (panels->curnode != panels->menu_pos && !strcmp (panels->panel[panels->curnode].cmd, PANEL_CMD_DISASSEMBLY)) {
```
Comment (XVilka) [inline libr/core/panels.c]: You forgot this one though.

File: libr/core/panels.c [hunk 2]
```diff
@@ -1672,7 +1695,7 @@ R_API int r_core_visual_panels(RCore *core, RPanels *panels) {
 		setRefreshAll (panels);
 		break;
 	case 'c':
-		if (!strcmp (panels->panel[panels->curnode].cmd, PANEL_CMD_STACK) || !strcmp (panels->panel[panels->curnode].cmd, PANEL_CMD_REGISTERS)) {
+		if (panels->curnode != panels->menu_pos && (!strcmp (panels->panel[panels->curnode].cmd, PANEL_CMD_STACK) || !strcmp (panels->panel[panels->curnode].cmd, PANEL_CMD_REGISTERS))) {
```
Comment (XVilka) [inline libr/core/panels.c]: And this one.

**Comment by XVilka** [comment]
Please also don't use `\` character where it is not strictily required - and it is not in this case.

**Comment by Vane11ope** [comment]
@XVilka ??? what do you mean. tbh i dont think this styling means anything anyway, but you asked to fold the thing so i folded then you telling me "do not use \".

**Comment by Vane11ope** [comment]
i shouldve actually argued at the first place. i dont see any reasonable reason for that styling. im getting old one back.

**Comment by radare** [comment]
The whole thibg must be refactored. This condition shouldnt exist at all. I wouldnt spent time cleaning this line

**Comment by Vane11ope** [comment]
but shouldnt be with this PR anyway i think.

**Comment by Vane11ope** [comment]
the refactoring is gonna be huge and i wouldnt prioritize that not right now.

**Comment by radare** [comment]
Totally agree

**Comment by XVilka** [comment]
Huh? You just write like 
```c
if (strncmp ... ) ||
 (strncmp ...) {
```
without `\` character. No refactoring or restyling required. Supersimple and I don't see any point of arguing with this.

**Comment by XVilka** [comment]
And it is not meaningless - it improves reading a lot, believe me.

**Comment by Vane11ope** [comment]
any editor would have the function to fold too long line. 

and they actually are not too long to me. i leave it to @radare

**Comment by XVilka** [comment]
Of course most of the editors can fold the line, but it will break the indent of the part that is folded.

**Comment by Vane11ope** [comment]
done.

File: libr/core/panels.c [hunk 3]
```diff
@@ -1643,8 +1643,10 @@ R_API int r_core_visual_panels(RCore *core, RPanels *panels) {
 			" sS   - step in / step over\n"
 			" uU   - undo / redo seek\n"
 			" pP   - seek to next or previous scr.nkey\n"
+			" o    - go/seek to given offset\n"
 			" nN   - create new panel with given command\n"
 			" q    - quit, back to visual mode\n"
+			" 0-9  - jump to the nth panel. n is the number you hit\n"
```
[the remark this answers is not in the document]
Comment (Vane11ope) [reply libr/core/panels.c]: there wasnt such thing even before me though.........

[the remark this answers is not in the document]
Comment (Vane11ope) [reply libr/core/panels.c]: currently this number thing does not break anything, but if you want me to change how it works then i will think about it.

[the remark this answers is not in the document]
Comment (XVilka) [reply libr/core/panels.c]: Can we use a modifier key for panel jumps? And make in-disasm jumps work too.

**Comment by XVilka** [comment]
One more thing - we start jumping to particular panel from 0. On many keyboards '0' located after '9'. I think it make sense to start counting from 1, to improve the user experience.

**Comment by Vane11ope** [comment]
makes sense. but probably not with this PR. mix too many things up.

**Pull request merged**